components/raftstore/src/store/worker/region.rs

```diff
@@ -365,6 +366,8 @@ where
         wb.write_opt(&wopts).unwrap_or_else(|e| {
             panic!("{} failed to save apply_snap result: {:?}", region_id, e);
         });
+        self.coprocessor_host
+            .on_apply_snapshot_committed(&region, peer_id, &snap_key, Some(&s));
```

Comment:
Why not move `post_apply_snapshot` here?

Reply:
`post_apply_snapshot` should taken place before, so region work can know the snapshot is already ingested by the underlying storage engine.